### ISSUE TITLE
Error on init: TypeError: Cannot read property 'identity' of undefined

### ISSUE THREAD

**Issue by marcelklehr** [balderdashy/waterline#745]
I get this error when calling waterline.initialize()
```
node_modules/waterline/lib/waterline/adapter/sync/strategies/alter.js:42
 = _.find(self.query.waterline.schema, {tableName: self.collection}).identity;

TypeError: Cannot read property 'identity' of undefined
    at afterDescribe (node_modules/waterline/lib/waterline/adapter/sync/strategies/alter.js:42:91)
    at bound (node_modules/waterline/node_modules/lodash/dist/lodash.js:957:21)
    at applyInOriginalCtx (node_modules/waterline/lib/waterline/utils/normalize.js:416:80)
    at wrappedCallback (node_modules/waterline/lib/waterline/utils/normalize.js:315:18)
    at _normalizeCallback.callback.success (node_modules/waterline/node_modules/node-switchback/lib/normalize.js:33:31)
    at _switch (node_modules/waterline/node_modules/node-switchback/lib/factory.js:35:28)
    at node_modules/sails-memory/lib/database.js:163:5
    at null._onTimeout (node_modules/sails-memory/lib/database.js:122:5)
    at Timer.listOnTimeout (timers.js:133:15)
```

**Comment by andrewjmead** [comment]
I'm getting that same error. I was able to debug it to be an issue with capitalization in the identity name. It would run once fine, then it would error out the second time. I updated the name to lower case and blasted the DB away and everything was fine.

```
// Name that did not work
identity: "linkStatus",

// Name that did work
identity: "linkstatus",

```

**Comment by emersion** [comment]
@andrewjmead Same problem, same solution. Thanks!

**Comment by andrewjmead** [comment]
@emersion Thanks for following up!

**Comment by jeremiahrhall** [comment]
The upper to lower case change worked for me as well.

**Comment by tjwebb** [comment]
Yea, by the way, when you don't manually set `identity`, sails automatically sets it to be lowercase.

**Comment by devinivy** [comment]
@tjwebb I thought identities were always lowercase–how do you "manually" set an identity with case?  If identities are always lowercase and that's not in the docs, let's add it.  Also, if identities should be lowercase, we should add a useful error message.

**Comment by tjwebb** [comment]
I don't see it documented anywhere but other github comments. Here's some relevant code: https://github.com/balderdashy/waterline/blob/master/lib/waterline.js#L116

We should add clarity on this behavior in https://github.com/balderdashy/waterline-docs/blob/master/models.md

**Comment by dmarcelino** [comment]
Added PR (balderdashy/waterline-docs#51), can someone double check?

**Comment by dmarcelino** [comment]
@tjwebb, @devinivy can I get one of you to bless balderdashy/waterline-docs#51 so we can close this issue? Thanks!

**Comment by dmarcelino** [comment]
Thanks @devinivy!

**Comment by devinivy** [comment]
Blessed!  But I am interested in removing the lowercase restriction some day.  When you read that in the docs, it sounds really bizarre.  In any case, adding this to the docs was the best short-term solution.

**Comment by japel** [comment]
+1 to remove lowercase restriction!
If I define a model mYoWeSoMeMoDeL.js I would like to call it the same way in waterline

**Comment by mlconnor** [comment]
the lower case constraint even seeps into the code.  now we have to use models.access_token instead of camel case accessToken.

+1 to remove restriction

**Comment by tjwebb** [comment]
@japel @mlconnor are you guys talking about the same issue as OP?

**Comment by GeorgeSapkin** [comment]
Any progress on this issue?

**Comment by gaurav21r** [comment]
@andrewjmead Its been many months but your solution is still very very helpful! @tjwebb @particlebanana In addition to the addition in the docs repo, I think we should have a better error message as well. `TypeError: Cannot read property 'identity' of undefined` is pretty obscure :stuck_out_tongue_closed_eyes:

**Comment by sailsbot** [comment]
Thanks for posting, @marcelklehr.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by japel** [comment]
Don't close this PLEASE! Don't mess with Lowercase, Uppercase, Camelcase, whatevercase! It should always be TheCaseIDecidedItToBe!

**Comment by devinivy** [comment]
@japel I agree with you!  Do you mind making a PR to the ROADMAP.md file that suggests we remove this restriction?
